myfitnesspal/client.py

```diff
@@ -795,3 +795,127 @@ def get_food_item_details(self, mfp_id: int) -> FoodItem:
+            "authenticity_token": authenticity_token,
+            "date": datetime.datetime.today().strftime("%Y-%m-%d"),
+            "food[brand]": brand,
+            "food[description]": description,
+            "weight[serving_size]": serving_size,
+            "servingspercontainer": "{}".format(servingspercontainer),
+            "nutritional_content[calories]": "{}".format(calories),
+            "nutritional_content[sodium]": "{}".format(sodium),
+            "nutritional_content[fat]": "{}".format(fat),
+            "nutritional_content[potassium]": "{}".format(potassium),
+            "nutritional_content[saturated_fat]": "{}".format(saturated_fat),
+            "nutritional_content[carbs]": "{}".format(carbs),
+            "nutritional_content[polyunsaturated_fat]": "{}".format(polyunsaturated_fat),
+            "nutritional_content[fiber]": "{}".format(fiber),
+            "nutritional_content[monounsaturated_fat]": "{}".format(monounsaturated_fat),
+            "nutritional_content[sugar]": "{}".format(sugar),
+            "nutritional_content[trans_fat]": "{}".format(trans_fat),
+            "nutritional_content[protein]": "{}".format(protein),
+            "nutritional_content[cholesterol]": "{}".format(cholesterol),
+            "nutritional_content[vitamin_a]": "{}".format(vitamin_a),
+            "nutritional_content[calcium]": "{}".format(calcium),
+            "nutritional_content[vitamin_c]": "{}".format(vitamin_c),
+            "nutritional_content[iron]": "{}".format(iron),
+            "food_entry[quantity]": "1.0",
+            "food_entry[meal_id]": "0",
+            "addtodiary": "no",
+            "preserve_exact_description_and_brand": "true",
+            "continue": "Save"
+        }
+        # Make entry public if requested, Hint: submit "sharefood": 0 also generates a public db entry, so only add
+        # "sharefood"" if really requested
+        if sharepublic:
+            data["sharefood"] = 1
+
+        submit4_url = parse.urljoin(self.BASE_URL_SECURE, self.SUBMIT_POST_PATH)
+        result = self.session.post(submit4_url, data, )
+        if result.status_code == 200:
+            document = lxml.html.document_fromstring(result.content.decode('utf-8'))
+            try:
+                if not document.xpath(
+                        # If list is empty there should be no error, could be replaced with assert
+                        "//*[@id='errorExplanation']/ul/li"):
+                    return True
+                    # print("No Error :)")
+                    # Would like to return FoodItem, but seems that it take
+                    # to long until the submitted food is available in the DB
+                    # return self.get_food_search_results("{} {}".format(brand, description))[0]
+                else:  # Error occurred
+                    error = document.xpath("//*[@id='errorExplanation']/ul/li")[0].text
+                    error = error.replace("Description ", "")
```

Comment:
Why do we need this? I took a look at `#errorExplanation` and didn't see any text of "Description ".

Reply:
Just for cosmetic reasons.  At least I got some errors where "Description " was placed before the error message. Just looked odd in the log/print out,

Reply:
I also get errors that start with "Description " e.g. 
`'Description An item with this brand and description already exists.'`
so this is a fine replacement to make I think.